locations/models.py

```diff
@@ -164,8 +163,7 @@ def main_image(self):
 
     def get_context(self, request):
         context = super().get_context(request)
-        locationtagindexset=LocationTagIndexPage.objects.live().sibling_of(self)
-        context["tagspage"]=locationtagindexset[0]
+        context['tagspage'] = LocationTagIndexPage.objects.live().child_of(LocationsIndexPage.objects.live().ancestor_of(self).first()).first()
```

Comment:
maybe a comment would be good: `# the first tag index page which is a child of the parent location index of this page`